### ISSUE TITLE
Compile build dependencies using cg_clif instead of cg_llvm

### ISSUE THREAD

**Issue by bjorn3** [bjorn3/rustc_codegen_cranelift#905]
This can be done by removing the `--target` argument in `cargo.sh`.

**Comment by bjorn3** [comment]
There is still an abi incompatibility for proc macro's.

**Comment by bjorn3** [comment]
The `__cxa_thread_atexit_impl` symbol has a different linkage compared to cg_llvm.

**Comment by bjorn3** [comment]
Patched the caller of `__cxa_thread_atexit_impl`. I am now stuck at `Bridge::cached_buffer` containing a wrong pointer.

**Comment by bjorn3** [comment]
[`Bridge`](https://github.com/rust-lang/rust/blob/36b1a9296cde2b773771710e9bbd608fd2eca35f/src/libproc_macro/bridge/mod.rs#L211) has `#[repr(C)]` and is too big to pass in two or less registers. This means that it is passed at the correct stack offset to the `extern "C" fn` (`Bridge::expand1::run`](https://github.com/rust-lang/rust/blob/36b1a9296cde2b773771710e9bbd608fd2eca35f/src/libproc_macro/bridge/client.rs#L391). cg_clif however expects a pointer to `Bridge` to be passed. The correct behavior is unfortunately not yet supported by Cranelift.

**Comment by bjorn3** [comment]
I got rustc bootstrapping using cg_clif. Unfortunately the rustc built using cg_clif crashes when you try to use it. So far I have been able to pin point that `make_input` returns an `io::Error`. When trying to `.to_string()` it it turns out that the `io::Error` has an invalid discriminant.

Rust fork: https://github.com/bjorn3/rust/commit/e5c309197bf97c12c5d0782b9300417a52c80f66 (branch cg_clif_subtree)
cg_clif commit: https://github.com/bjorn3/rustc_codegen_cranelift/commit/f7eb36070c602f0dccca26ccec83f1c003cb27dd (branch wip_proc_macro_fixes, clone it to `src/rustc_codegen_cranelift` in the rust source repo)
Cranelift commit: https://github.com/bjorn3/wasmtime/commit/2d60da466dc375b0cb08ad62e49879927f176619 (pr bytecodealliance/wasmtime#1559, clone it to the parent dir of the rust repo, or edit Cargo.toml of the rust repo to point to it in the patch section)

config.toml:
```toml
[rust]
codegen-backends = ["cranelift"]
deny-warnings = false
```
